### ISSUE TITLE
Neural Networks

### ISSUE THREAD

**Issue by ingo** [DuckieOverload/transferlearningsophon#3]
## Goal
Build intuition for how neural networks work

## Watch
3Blue1Brown: Neural networks  
https://www.3blue1brown.com/topics/neural-networks

You only need to watch these three:
- “But what is a Neural Network?”
- “Gradient descent, how neural networks learn”
- “Backpropagation, intuitively”

## Tasks
- [ ] Watch the video
- [ ] Answer:
  - What is a neural network, and how does it transform input data into output predictions?
  - What role do weights and biases play in a neural network, and why do we adjust them during training?
  - What is gradient descent, and why is it important for training neural networks?
  - Why do we need backpropagation in order to train a neural network?
  - Why are activation functions (like ReLU or sigmoid) used between layers instead of just having linear transformations?

## Repo Pairing
- Locate model definitions (MLP, ParticleTransformer)
- Identify stacked layers in the model

## Done When
You can describe a neural network as a function rather than a black box.

**Comment by DuckieOverload** [comment]
1. A neural network is a model inspired by the human brain, and consists of layers of interconnected notes or "neurons." It transforms data from the input to the output by passing through these layers. Each layer performs a mathematical calculation that extracts complex patterns and creates a final result at the end.
2. Weights and biases are adjustable "knobs" on a model that determine how much influence any given input has on the final result. we adjust them because tuning them can lead to better and more accurate results.
3. Gradient descent is the optimization algorithm used to reduce errors. The algorithm calculates the "slope" of the error and takes small steps in the steepest downward direction. It is important because it can update lots of weights and biases easily while doing the algorithm.
4. Because it tells the neural network how to improve. It lets the model know which "knob" to turn and by how much, so that it learns from its mistakes.
5. They are added to give the model non-linearity. Without them, the model would be linear and therefore could not learn complex patterns.

**Comment by ingo** [comment]
Nice answers.
